vault/resource_terraform_cloud_secret_creds.go

```diff
@@ -0,0 +1,154 @@
+			},
+			"token": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+				Description: "Terraform Token provided by the Vault backend",
+			},
+			"token_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the Terraform Token provided",
+			},
+			"organization": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "Name of the Terraform Cloud or Enterprise organization",
+			},
+			"team_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ID of the Terraform Cloud or Enterprise team under organization (e.g., settings/teams/team-xxxxxxxxxxxxx)",
+			},
+		},
+	}
+}
+
+func createTerraformCloudSecretCredsResource(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	backend := d.Get("backend").(string)
+	role := d.Get("role").(string)
+	path := fmt.Sprintf("%s/creds/%s", backend, role)
+
+	secret, err := client.Logical().Read(path)
+	if err != nil {
+		return fmt.Errorf("error reading from Vault: %s", err)
+	}
+	log.Printf("[DEBUG] Read %q from Vault", path)
+
+	if secret == nil {
+		return fmt.Errorf("no role found at %q", path)
+	}
+
+	token := secret.Data["token"].(string)
+	if token == "" {
+		return fmt.Errorf("token is not set in response")
+	}
+	tokenId := secret.Data["token_id"].(string)
+
+	organization := secret.Data["organization"]
+	teamId := secret.Data["team_id"]
+
+	d.SetId(tokenId)
+
+	if secret.LeaseID != "" {
+		d.Set("lease_id", secret.LeaseID)
+	} else {
+		d.Set("lease_id", "")
+	}
+
+	d.Set("token", token)
+	d.Set("token_id", tokenId)
+	d.Set("organization", organization)
+	d.Set("team_id", teamId)
+
+	return nil
+}
+
+func deleteTerraformCloudSecretCredsResource(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	leaseId := d.Get("lease_id").(string)
+
+	organization := d.Get("organization")
+	teamId := d.Get("team_id")
+
+	if organization == "" && teamId == "" {
+		err := client.Sys().Revoke(leaseId)
+		if err != nil {
+			return fmt.Errorf("error revoking token from Vault: %s", err)
+		}
+		log.Printf("[DEBUG] Revoked lease: %q from Vault, removing user token %s from state", leaseId, d.Id())
+		d.SetId("")
+	} else {
+		log.Printf("[DEBUG] No lease to revoke for Team/Org tokens")
+	}
+	return nil
+}
+
+func updateTerraformCloudSecretCredsResource(d *schema.ResourceData, meta interface{}) error {
+	err := deleteTerraformCloudSecretCredsResource(d, meta)
+	if err != nil {
+		return fmt.Errorf("previous token not revoked: %s", err)
+	}
+	err = createTerraformCloudSecretCredsResource(d, meta)
+	return err
+}
+
+func readTerraformCloudSecretCredsResource(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*api.Client)
+	leaseId := d.Get("lease_id")
+	organization := d.Get("organization")
+	teamId := d.Get("team_id")
+
+	if organization == "" && teamId == "" {
```

Comment:
Same here, you can now lease perform the lookup if `leaseId != ""` (maybe even get rid of `organization == "" && teamId == ""` since it doesn't really depend on that).

Reply:
Fixed in latest commit.